src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -456,6 +455,65 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly nonMutationMolecularProfileDataAvailability = remoteData<{[molecularProfileId:string]:{[entrezGeneId:number]:number}}>({
+        await:()=>[
+            this.molecularProfilesInStudies,
+            this.studyToDataQueryFilter,
+            this.genes
+        ],
+        invoke:async()=>{
+            const ret:{[molecularProfileId:string]:{[entrezGeneId:number]:number}} = {};
+            const promises = [];
+            const studyToDataQueryFilter = this.studyToDataQueryFilter.result!;
+            for (const profile of this.molecularProfilesInStudies.result!) {
+                if (profile.molecularAlterationType === AlterationTypeConstants.MUTATION_EXTENDED) {
+                    continue;
+                }
+                for (const gene of this.genes.result!) {
+                    const molecularDataFilter = {
+                        entrezGeneIds: [gene.entrezGeneId],
+                        ...studyToDataQueryFilter[profile.studyId]
+                    } as MolecularDataFilter;
+                    const molecularProfileId = profile.molecularProfileId;
+                    const projection = "META";
+                    promises.push(client.fetchAllMolecularDataInMolecularProfileUsingPOSTWithHttpInfo({
+                        molecularProfileId,
+                        molecularDataFilter,
+                        projection
+                    }).then(function(response: request.Response) {
+                        ret[molecularProfileId] = ret[molecularProfileId] || {};
+                        ret[molecularProfileId][gene.entrezGeneId] = parseInt(response.header["total-count"], 10);
+                    }));
+                }
+            }
+            await Promise.all(promises);
+            return ret;
+        }
+    });
+
+    readonly nonMutationMolecularProfilesWithData = remoteData<{[entrezGeneId:number]:MolecularProfile[]}>({
+        await:()=>[
+            this.genes,
+            this.molecularProfilesInStudies,
+            this.nonMutationMolecularProfileDataAvailability
+        ],
+        invoke:()=>{
+            const ret:{[entrezGeneId:number]:MolecularProfile[]} = {};
+            const molecularProfilesInStudies = this.molecularProfilesInStudies.result!;
+            const nonMutationMolecularProfileDataAvailability = this.nonMutationMolecularProfileDataAvailability.result!;
+            for (const gene of this.genes.result!) {
+                ret[gene.entrezGeneId] = molecularProfilesInStudies.filter(p=>{
+                    if (p.molecularProfileId in nonMutationMolecularProfileDataAvailability) {
```

Comment:
i think you could just return nonMutationMolecularProfileDataAvailability[p.molecularProfileId]===true ?

Reply:
that's not the data structure but you're right it would be clearer to return nonMutation... > 0

Reply:
actually I think we want to keep this check but I'll add a comment to explain it